src/pp-annotator/form/AnnotationForm.tsx

```diff
@@ -54,77 +63,187 @@ export default class AnnotationForm extends React.Component<
 
     this.handleInputChange = this.handleInputChange.bind(this);
     this.onSave = this.onSave.bind(this);
+    this.executeSave = this.executeSave.bind(this);
     this.onCancel = this.onCancel.bind(this);
   }
 
+  public componentWillUpdate(_nextProps: IAnnotationFormProps, nextState: Partial<IAnnotationFormState>) {
+    // Whenever the field has changed, eradicate the error message
+    if(nextState.referenceLink) {
+      nextState.referenceLinkError = '';
+    }
+    if(nextState.referenceLinkTitle) {
+      nextState.referenceLinkTitleError = '';
+    }
+  }
+
+  public saveButtonClass() {
+    const priorityToClass = {
+      [AnnotationPriorities.NORMAL]: 'priority-normal',
+      [AnnotationPriorities.WARNING]: 'priority-warning',
+      [AnnotationPriorities.ALERT]: 'priority-alert',
+    };
+    return priorityToClass[this.state.priority || AnnotationPriorities.NORMAL];
+  }
+
+  // A modal displayed when user tries to save the form with comment field empty
+  public renderNoCommentModal() {
+      this.noCommentModal = (
+          <Modal
+              open={this.state.noCommentModalOpen}
+              size="mini"
+          >
+            <Modal.Content>
+              Czy na pewno chcesz dodać przypis bez treści?
+            </Modal.Content>
+            {/* Action buttons style from semantic-ui, probably temporary */}
+            <Modal.Actions>
+              <button
+                  className="ui button negative"
+                  onClick={() => this.setState({noCommentModalOpen: false})}
+              >
+                Anuluj
+              </button>
+              <button
+                  className="ui button"
+                  onClick={this.executeSave}
+              >
+                Zapisz
+              </button>
+            </Modal.Actions>
+          </Modal>
+      );
+      return this.noCommentModal;
+  }
+
   public render() {
       const {
           priority,
           comment,
           referenceLink,
-          referenceLinkTitle
+          referenceLinkError,
+          referenceLinkTitle,
+          referenceLinkTitleError,
       } = this.state;
 
     return (
-      <div className="pp-widget">
-        <div className="pp-editor-head-bar">
-            <div className={"pp-editor-priority" + (priority=='NORMAL' ? ' selected' : '')}>
-                przypis
+        <div className="pp-widget">
+          <div className="pp-editor-head-bar">
+            <label className="priority-header"> Co dodajesz? </label>
+            <Popup
+                on="click"
+                hideOnScroll
+                trigger={<div className="priority-help"> <i className="help circle icon"></i> </div>}
+                flowing
+                hoverable
+            >
+              {/*TODO just an instruction stub*/}
+              <Grid centered divided columns={3}>
```

Comment:
I'm not sure if we should use semantic ui components for layout (ui styling is fine), isn't flexbox enough?

Reply:
I took Grid from an example at hand. This Popup's content will be changed anyway (I probably should have left a comment on that), I'll remember your suggestion when I'm rewriting it.